### DISCUSSION TITLE
Transformation for infantry

### DISCUSSION THREAD

**Discussion by 8abs** [Phobos-developers/Phobos#1023] [Ideas and feedback]
### First check

- [X] The suggestion is **NOT** related to functionality of Ares, HAres, Kratos or any other YR engine extension.
- [X] The suggested feature / enhancement is **NOT** implemented in Ares/Phobos yet and the existing possibilites are not viable or good enough to achieve the wanted result.
- [X] I agree to elaborate the details if requested and provide thorough testing if the feature is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I thought out how the feature / enhancement should work and described it in the most detailed and clear way.

### Description

Ares has a "IsSimpleDeployer=<boolen>" , so that a vehicle can transform into another vehicle only by click "D".
Of course, we can use deployfire and InfDeathAnim and makeinfantry to make a infantry become another also only by a click in game. But :
(1)  It's too troublesome
 to define a lot of animations & warheads.
(2) The essence of this is killing the infantry itself and create a new unit, if this infantry is a key unit (like boss or hero in some possible combat or multi-player game mode), this kind of transformation would directly lead to failure.

So there should be some INI code that can make infantry some other infantry directly. And the transformation don't take into death count and would lead the player to failure if this is the last "KeepAlive" unit.

### INI code

```ini
[SomeInfantryType]
CanTrans=yes
TransInto=<some infantry type>
TransConditions=Deploy      ;Options can be:Deploy,Death,Hit,MindControl   ;Conditions for transformation
```

**Comment by kingoni1990** [comment]
> ### First check
> * [x]  The suggestion is **NOT** related to functionality of Ares, HAres, Kratos or any other YR engine extension.
> * [x]  The suggested feature / enhancement is **NOT** implemented in Ares/Phobos yet and the existing possibilites are not viable or good enough to achieve the wanted result.
> * [x]  I agree to elaborate the details if requested and provide thorough testing if the feature is implemented.
> * [x]  I added a very descriptive title to this issue.
> * [x]  I used the GitHub search and read the issue list to find a similar issue and didn't find it.
> * [x]  I thought out how the feature / enhancement should work and described it in the most detailed and clear way.
> 
> ### Description
> Ares has a "IsSimpleDeployer=" , so that a vehicle can transform into another vehicle only by click "D". Of course, we can use deployfire and InfDeathAnim and makeinfantry to make a infantry become another also only by a click in game. But : (1) It's too troublesome to define a lot of animations & warheads. (2) The essence of this is killing the infantry itself and create a new unit, if this infantry is a key unit (like boss or hero in some possible combat or multi-player game mode), this kind of transformation would directly lead to failure.
> 
> So there should be some INI code that can make infantry some other infantry directly. And the transformation don't take into death count and would lead the player to failure if this is the last "KeepAlive" unit.
> 
> ### INI code
> ```ini
> [SomeInfantryType]
> CanTrans=yes
> TransInto=<some infantry type>
> TransConditions=Deploy      ;Options can be:Deploy,Death,Hit,MindControl   ;Conditions for transformation
> ```
> 
> ### Additional context
> _No response_

hope effect with TANK, AIRCRAFT
